README.md

```diff
@@ -8,15 +8,51 @@ of Minecraft.
 
 ## Requirements
 
-### For Running From Source
+### Running from Source
+In order to run the Amulet Editor from source, you will need to install the following packages and follow the steps to install below:
 - Python 3.7.0+
 - Numpy
 - pyglet
-- NBT
+- [Amulet_NBT](https://github.com/Amulet-Team/Amulet-NBT)
+
+The steps below will install everything needed for Amulet (Instructions for installing Python 3.7 provided when available)
+
+__Note: Make sure to install the `venv` module if it does not come pre-installed with python__
 
-Shortcut: `pip install -r requirements.txt`
+#### Windows
+1. Clone the project using `git clone https://github.com/Amulet-Team/Amulet-Map-Editor`
+2. Change the working directory to be the directory created when cloning (Ex: `cd Amulet-Map-Editor`)
+3. Set up a python virtual environment (run the following commands in the command line prompt)
+   1. To create the virtual environment: `python -m venv ENV`
+   2. To activate the environment: `.\ENV\scripts\activate.bat`
+4. Install the requirements using `pip install -r requirements.txt`
+5. To format your files automatically before committing changes, use `pre-commit install`
+
+#### OS X - In Progress
+1. Clone the project using `git clone https://github.com/Amulet-Team/Amulet-Map-Editor`
+2. Change the working directory to be the directory created when cloning (Ex: `cd Amulet-Map-Editor`)
+3. Set up a python virtual environment (run the following commands in the Terminal app)
+   1. To create the virtual environment: `python3.7 -m venv ENV`
+      - Depending on how you installed python 3.7, this might instead be: `py -3.7 -m venv ENV`
+   2. To activate the environment: `source /ENV/bin/activate`
+4. Install the requirements using `pip install -r requirements.txt`
+5. To format your files automatically before committing changes, use `pre-commit install`
+
+#### Linux - In Progress
+1. Clone the project using `git clone https://github.com/Amulet-Team/Amulet-Map-Editor`
+2. Change the working directory to be the directory created when cloning (Ex: `cd Amulet-Map-Editor`)
+3. Install python 3.7 if it's not already installed (run the following commands in your terminal of choice; these mary vary depending on your linux distribution)
+   1. `sudo apt-get update`
```

Comment:
You should point out that this command is for Debian and distributions based on it such as Ubuntu or Linux Mint and it is optional as it only updates package catalogue. It is recommended to run this command only if was not run recently as it downloads some data from the internet during that.